src/foam/dao/index/ProxyIndex.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao.index;
+
+import foam.core.FObject;
+import foam.dao.Sink;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+
+public class ProxyIndex
+    implements Index
+{
+  protected Index delegate_;
+  private boolean delegateIsSet_ = false;
```

Comment:
This is done in modelled objects to support default values, but isn't needed here. Just remove all of the delegateIsSet_ support.

Reply:
Done